STM32/FLASH_readwrite/Src/HAL_H7_otp.c

```diff
@@ -20,54 +20,58 @@
 // to OTP memory. This part of the memory is therefore not allowed to write to during
 // normal FLASH storing.
 #define FLASH_OTP_BASE		0x08080000
-#define OTP_FLASH_SECTOR	0x04
-
-uint32_t OTP_CLEAR_ADDRESS = 0xFFFFFFFF;
-
+#define OB_WRP_SECTOR       OB_WRP_SECTOR_4
 #define FLASH_WORD_SIZE		32 // 32 bytes
 
-// This method can change the write-protection of a sector of BANK 1.
-// The documentation for this may be found in reference manual RM0433
-// section 4.5.1 and 4.9.16.
-static HAL_StatusTypeDef changeWriteProtectedSector(bool enable)
-{
-	if (!enable && ((FLASH->OPTCR & 0x01) == 0U))
-	{
-		// Remove write protection of 'OTP' sector
-		FLASH->WPSN_PRG1 |= (1UL << OTP_FLASH_SECTOR);
-
-		if (((FLASH->WPSN_PRG1 >> 4) & 0x01) != 0x01)
-			return HAL_ERROR;
-
-	}
-	else if (enable && ((FLASH->OPTCR & 0x01) == 0U))
-	{
-		// Enable write protection of 'OTP' sector
-		FLASH->WPSN_PRG1 &= ~(1UL << OTP_FLASH_SECTOR);
-
-		if (((FLASH->WPSN_PRG1 >> 4) & 0x01) != 0U)
-			return HAL_ERROR;
-	}
-	return HAL_OK;
-}
 
 // Check whether OTP data is already programmed on board.
-static int isOTPAvailable()
+static int isOTPAvailable(FLASH_OBProgramInitTypeDef *OB)
 {
-    volatile uint32_t* p = (uint32_t*) FLASH_OTP_BASE;
-    if (*p == OTP_CLEAR_ADDRESS)
-    	return 0;
+    if (OB->WRPSector & OB_WRP_SECTOR)
+        return 0;
 
-    // all OTP sectors is free
+    // OTP sector is write-protected
     return 1;
 }
 
+// Remove existing OTP data 
+int removeOTP(FLASH_OBProgramInitTypeDef *OB, const BoardInfo *boardInfo)
+{
+    if (HAL_FLASH_Unlock() != HAL_OK)
+        return OTP_WRITE_FAIL;
+
+    // Unlock Flash OPT registry that allows removal of write protection of FLASH section
```

Comment:
OTP

Reply:
I actually meant option bytes so double wrong from my part. The comments are updated now.